src/ElvClient.js

```diff
@@ -975,6 +975,75 @@ class ElvClient {
     ]))}`;
   }
 
+
+  /**
+   * Build a signed message (JSON) using the current signer.
+   * Signed messages have a similar format to signed access tokens and they include the message itself
+   * such that they can be both verified and decoded by the receiving entity.
+   *
+   * Messages can be encoded and signed using different methods, and the encoding and signature types
+   * are described in the header of the resulting signed message blob.
+   *
+   * Note this type of message can not be verified and decoded on chain.
+   *
+   * @methodGroup Authorization
+   * @namedParams
+   * @param {string} messasge - A JSON object representing the message to sign
+   */
+  async CreateSignedMessageJSON({
+      message
+  }) {
+
+    // Only one kind of signature supported currently
+    const type = "mje_" // JSON message, EIP192 signature
+
+    const msg = JSON.stringify(message);
+    const signature = await client.PersonalSign({message: msg, addEthereumPrefix: true});
+    return `${type}${Utils.B58(
+      Buffer.concat([
+        Buffer.from(signature.replace(/^0x/, ""), "hex"),
+        Buffer.from(msg)
+      ])
+    )}`;
+  }
+
+  /**
+   * Verify and decode a signed message (JSON).
+   *
+   * @methodGroup Authorization
+   * @namedParams
+   * @param {string} signedMessage - a signed message as created by CreateSignedMessageJSON
+   */
+  async DecodeSignedMessageJSON({
+    signedMessage
+  })  {
+    const type = signedMessage.slice(0,4);
+    let res = {};
+    switch(type) {
+        case "mje_":
+            const msgBytes = Utils.FromB58(signedMessage.slice(4));
+            const signature = msgBytes.slice(0, 65);
+            const msg = msgBytes.slice(65);
+            const obj = JSON.parse(msg);
+
+            // TODO - Verify signature and extract signer
+            //
+            signerAddr = null;
+
+            res = {
+                type: type,
+                obj: obj,
```

Comment:
return value json should be called `message`, and let's also call addr `address`, more consistent with the rest of the client

Reply:
In authd we have this:
```
type DecodedMessage struct {
	Type      string                 `json:"type"`
	Obj       map[string]interface{} `json:"obj"`
	Signature string                 `json:"signature"`
}
```

Also, signer not included.

Reply:
change it in the authd PR, branch `add-nft-claim-entitlement-action`

it also needs to be updated in
- media-wallet PR
- elv-sample-custom-app-purchase

but we should just finalize it here and pull out the duplicated code in them anyway, so, fine to change formats